### PR TITLE
Call-site inline single dispatch calls to JS methods

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
Still needs a test. IIUC the best we can do (given the infrastructure we have) is test that the compiler generates an Apply node with the correct flags.

File: compiler/src/main/scala/org/scalajs/nscplugin/GenJSExports.scala [hunk 1]
```diff
@@ -354,8 +369,19 @@ trait GenJSExports[G <: Global with Singleton] extends SubComponent {
 
       val overloads = alts.map(new ExportedSymbol(_, static))
 
-      val (formalArgs, restParam, body) =
-        genOverloadDispatch(jsName, overloads, jstpe.AnyType)
+      val (formalArgs, restParam, body) = {
+        if (allowCallsiteInlineSingle && overloads.size == 1) {
```
Comment (sjrd) [inline compiler/src/main/scala/org/scalajs/nscplugin/GenJSExports.scala:373]: Maybe we should take this fast path even when `!allowCallsiteInlineSingle`, and use `inline = allowCallsiteInlineSingle` a few lines below?

File: compiler/src/main/scala/org/scalajs/nscplugin/GenJSExports.scala [hunk 2]
```diff
@@ -316,9 +323,17 @@ trait GenJSExports[G <: Global with Singleton] extends SubComponent {
         } else {
           val formalArgsRegistry = new FormalArgsRegistry(1, false)
           val (List(arg), None) = formalArgsRegistry.genFormalArgs()
-          val body = genOverloadDispatchSameArgc(jsName, formalArgsRegistry,
-              alts = setters.map(new ExportedSymbol(_, static)), jstpe.AnyType,
-              paramIndex = 0)
+
+          val body = {
+            if (allowCallsiteInlineSingle && setters.size == 1) {
+              genApplyForSym(formalArgsRegistry, setters.head, static, inline = true)
```
Comment (sjrd) [inline compiler/src/main/scala/org/scalajs/nscplugin/GenJSExports.scala:329]: Same here?
```suggestion
            if (setters.size == 1) {
              genApplyForSym(formalArgsRegistry, setters.head, static, inline = allowCallsiteInlineSingle)
```

**Review by sjrd — COMMENTED** [review]
Looks good so far.

File: compiler/src/main/scala/org/scalajs/nscplugin/GenJSExports.scala [hunk 3]
```diff
@@ -865,12 +865,13 @@ trait GenJSExports[G <: Global with Singleton] extends SubComponent {
         if (sym.isClassConstructor)
           genNew(currentClassSym, sym, args)
         else if (sym.isPrivate)
-          ensureResultBoxed(genApplyMethodStatically(receiver, sym, args), sym)
+          ensureResultBoxed(genApplyMethodStatically(receiver, sym, args, inline = inline), sym)
         else
-          ensureResultBoxed(genApplyMethod(receiver, sym, args), sym)
+          ensureResultBoxed(genApplyMethod(receiver, sym, args, inline = inline), sym)
```
Comment (sjrd) [inline compiler/src/main/scala/org/scalajs/nscplugin/GenJSExports.scala:898]: These are in the wrong commit.

**Review by sjrd — CHANGES_REQUESTED** [review]
One wrong-commit comment, but otherwise LGTM.

**Review by sjrd — APPROVED** [review]

**Pull request merged**